core_lib/src/util/fileformat.cpp

```diff
@@ -63,3 +64,47 @@ QString retrieveProjectNameFromTempPath(const QString& path)
     //qDebug() << tokens;
     return tokens[0];
 }
+
+QString detectFormatByFileNameExtension(const QString& fileName)
+{
+    QMap<QString, QString> extensionMapping
+        {
+            { "png",  "PNG" },
+            { "jpg" , "JPG" },
+            { "jpeg", "JPG" },
+            { "tif",  "TIF" },
+            { "tiff", "TIF" },
+            { "bmp",  "BMP" },
+            { "mp4",  "MP4" },
+            { "avi",  "AVI" },
+            { "gif",  "GIF" },
+            { "webm", "WEBM" },
+            { "apng", "APNG" },
+        };
+
+    QString extension = fileName.mid(fileName.lastIndexOf(".") + 1).toLower();
+    if (!fileName.contains(".") || !extensionMapping.contains(extension))
+    {
+        return QString();
+    }
+    return extensionMapping[extension];
+}
+
+bool isMovieFormat(const QString& format)
+{
+    QMap<QString, bool> formatMapping
+        {
+            { "PNG", false },
+            { "JPG", false },
+            { "TIF", false },
+            { "BMP", false },
+            { "MP4",  true },
+            { "AVI",  true },
+            { "GIF",  true },
+            { "WEBM", true },
+            { "APNG", true },
+        };
+
+    Q_ASSERT(formatMapping.contains(format));
+    return formatMapping[format];
```

Comment:
The idea is that the keys in this formatMapping correspond exactly to the values of the extensionMapping above. Poor man’s constants, basically. That’s essentially what I wanted to express with the assertion: If the exact key does not exist in the mapping, it doesn’t just indicate an unknown format, it indicates an error in the program logic. But you are absolutely correct that, as it is now, it is a very ugly solution; one inherited [from the original command line exporter](https://github.com/pencil2d/pencil/pull/1545/files/984da01a00d768e3c1331e4b6d91cd3248c9f532#diff-a68bbd02ff27204e90b2bcf21c852b2a42bd7921db32fb0169dca0661e388820L257-L295) prior to this commit. The reason I didn’t do anything about it is that our handling of non-native file formats in general is currently not very centralised, with custom code in several different places (for example here, in Object, MovieExporter, ExportMovieDialog…). Cleaning that up is on my todo list already, so instead of fixing this particular piece of code right now, I’d rather address the file format situation as a whole separately from this PR. Would that be okay with you or would you still prefer to have this particular part cleaned up sooner rather than later?

Reply:
That sounds fine to me, in that case then I have no further issues, LGTM.